src/features/import/hooks/useColumn.ts

```diff
@@ -56,32 +57,38 @@ export default function useColumn(orgId: number) {
   const nativeFieldsOptions: Option[] = Object.values(NATIVE_PERSON_FIELDS)
     .filter((fieldSlug) => fieldSlug != 'id' && fieldSlug != 'ext_id')
     .map((fieldSlug) => ({
+      disabled: false,
       label: globalMessages.personFields[fieldSlug](),
       value: `field:${fieldSlug}`,
     }));
 
-  const customFieldsOptions: Option[] = customFields
-    .filter(
-      (field) => field.organization.id == orgId || field.org_write == 'suborgs'
-    )
-    .map((field) => {
-      if (field.type === CUSTOM_FIELD_TYPE.DATE) {
-        return {
-          label: field.title,
-          value: `date:${field.slug}`,
-        };
-      } else if (field.type == CUSTOM_FIELD_TYPE.ENUM && field.enum_choices) {
-        return {
-          label: field.title,
-          value: `enum:${field.slug}`,
-        };
-      } else {
-        return {
-          label: field.title,
-          value: `field:${field.slug}`,
-        };
-      }
-    });
+  const customFieldsOptions: Option[] = customFields.map((field) => {
+    if (field.type === CUSTOM_FIELD_TYPE.DATE) {
+      return {
+        disabled: !(
+          field.organization.id == orgId || field.org_write == 'suborgs'
+        ),
```

Comment:
Maybe this logic could be moved to a separate variable over the if statements?

On top of moving it, I would personally also refactor it. This is just my personal preference, but I can find complex logic like this difficult to read. In situations like this, I find it makes it easier to name each condition (as a variable), e.g.:

```ts
const belongsToCurrentOrg = field.organization.id == orgId;
const suborgsCanWrite = field.org_write == 'suborgs';
const currentOrgCanWrite = belongsToCurrentOrg || suborgsCanWrite;
const readOnly = !currentOrgCanWrite;
```

In that case, this could be:

```suggestion
        disabled: readOnly,
```

Reply:
I agree, less repetitive too :)